packages/zowe-explorer/src/trees/dataset/DatasetFSProvider.ts

```diff
@@ -660,38 +691,50 @@ export class DatasetFSProvider extends BaseProvider implements vscode.FileSystem
                     entry.mtime = Date.now();
                     entry.size = content.byteLength;
                     entry.inDiffView = true;
+
+                    parent.entries.set(basename, entry);
```

Comment:
Please correct me if I'm missing something here, but since entry is already in `parent.entries`, updating properties on `entry` should propagate without calling `parent.entries.set` (as entry is a reference to the object in `parent.entries`).

Reply:
I agree with that premise. However... 😓 
I wasn't seeing that behavior during unit testing (once they are executed in isolation). That indicates that without setting the entry (again), the original entry in the diff could be left untouched (even after the user has made changes to it)

Reply:
> I wasn't seeing that behavior during unit testing (once they are executed in isolation)

I think that could be an issue with how the test objects are initialized, and since this is a nitpick I don't want to suggest any changes for this one at this time. Thanks for the context though 😋

Reply:
I think in the context of the tests, this call is necessary because the entry doesn't persist between test cases - as you've mentioned, the tests are running in isolation. Some of the mock objects are pre-defined at the top of the file, so I am wondering if there's some pollution going on between the test cases..

But in a real-world scenario with the FSP, the entry is a reference to the object stored in the `entries` map - so that when entry is changed, the changes are reflected for that entry in the map without needing to do another `set`.

Again, just nitpicking and this won't impact the behavior in production - I just wanted to see if we could avoid it 😋